src/main/features/claimant-response/guards/allClaimantResponseTasksCompletedGuard.ts

```diff
@@ -0,0 +1,33 @@
+import * as express from 'express'
+
+import { Draft } from '@hmcts/draft-store-client'
+import { Logger } from '@hmcts/nodejs-logging'
+import { Claim } from 'claims/models/claim'
+import { Paths } from 'claimant-response/paths'
+import { TaskListBuilder } from 'claimant-response/helpers/taskListBuilder'
+import { DraftClaimantResponse } from 'claimant-response/draft/draftClaimantResponse'
+
+const logger = Logger.getLogger('router/claimant-response/check-and-send')
+
+export class AllClaimantResponseTasksCompletedGuard {
+
+  static async requestHandler (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+    try {
+      const draft: Draft<DraftClaimantResponse> = res.locals.claimantResponseDraft
+      const claim: Claim = res.locals.claim
+
+      const allTasksCompleted: boolean = TaskListBuilder
+        .buildRemainingTasks(draft.document, claim).length === 0
+
+      if (allTasksCompleted) {
+        return next()
+      }
+
+      logger.debug('State guard: claim check and send page is disabled until all tasks are completed - redirecting to task list')
```

Comment:
remove claim word from here, the logger name should be enough to know which check and send page